pkg/gpu/testutil/samplebins.go

```diff
@@ -26,27 +31,76 @@ const (
 	CudaSample SampleName = "cudasample"
 )
 
-// RunSample executes the sample binary and returns the command. Cleanup is configured automatically
-func RunSample(t *testing.T, name SampleName) (*exec.Cmd, error) {
+// DockerImage represents the Docker image to use for running the sample binary.
+type DockerImage string
+
+const (
+	// MinimalDockerImage is the minimal docker image, just used for running a binary
+	MinimalDockerImage DockerImage = "alpine:3.20.3"
+)
+
+func getBuiltSamplePath(t *testing.T, sample SampleName) string {
 	curDir, err := testutil.CurDir()
 	require.NoError(t, err)
 
-	sourceFile := filepath.Join(curDir, "..", "testdata", string(name)+".c")
-	binaryFile := filepath.Join(curDir, "..", "testdata", string(name))
+	sourceFile := filepath.Join(curDir, "..", "testdata", string(sample)+".c")
+	binaryFile := filepath.Join(curDir, "..", "testdata", string(sample))
 
 	builtBin, err := buildCBinary(sourceFile, binaryFile)
 	require.NoError(t, err)
 
+	return builtBin
+}
+
+// RunSample executes the sample binary and returns the command. Cleanup is configured automatically
+func RunSample(t *testing.T, name SampleName) (*exec.Cmd, error) {
+	builtBin := getBuiltSamplePath(t, name)
+
+	log.Debugf("Running sample binary %s with command %v", name, builtBin)
 	cmd := exec.Command(builtBin)
 	t.Cleanup(func() {
 		if cmd.Process != nil {
 			_ = cmd.Process.Kill()
 		}
 	})
-	err = cmd.Start()
+	err := cmd.Start()
 	require.NoError(t, err)
 
-	log.Infof("Running sample binary %s with PID %d", name, cmd.Process.Pid)
-
 	return cmd, nil
 }
+
+// RunSampleInDocker executes the sample binary in a Docker container and returns the PID of the main container process, and the container ID
+func RunSampleInDocker(t *testing.T, name SampleName, image DockerImage) (int, string, error) {
```

Comment:
nit: consider returning a struct that contains the pid and cid

Reply:
I think for now it's simpler to return two values, specially as this is test code.